addons/event/models/event_mail.py

```diff
@@ -190,13 +173,51 @@ def execute(self):
                 })
         return True
 
+    def _execute_after_subscription(self, registration_limit=1000):
+        """ 'after_sub' scheduler management. It currently does two main things
+          * generate missing 'event.mail.registrations' which are scheduled
+            communication linked to registrations;
+          * launch registration-based communication, splitting in batches as
+            it may imply a lot of computation. When having more than given
+            limit to handle, schedule another call of cron to avoid having to
+            wait another cron interval check;
+        """
+        # fillup on subscription lines
+        if self.env.context.get('event_mail_registration_ids'):
+            new_registrations = self.env['event.registration'].search([
+                ('id', 'in', self.env.context['event_mail_registration_ids']),
+                ('event_id', '=', self.event_id.id),
+            ])
+        else:
+            new_registrations = self.event_id.registration_ids.filtered_domain(
+                [('state', 'not in', ('cancel', 'draft'))]
+            )
```

Comment:
This one also seems quetionnable in terms of performance.

We are basically fetching ALL registrations, then filtering on cancel/draft, then further filtering based on "mail_registration_ids.registration_id".

Could probably benefit a lot from a single search.

The one above could also benefit from it, would probably build a base_domain based on the context and then search.